gclient/tests/upload.rs

```diff
@@ -61,16 +50,65 @@ async fn harmless_upload() -> Result<()> {
         .collect();
 
     // Checking that all upload program calls succeed in batch.
-    assert_eq!(PATHS.len(), mids.len());
+    assert_eq!(codes_len, mids.len());
 
     // Checking that all batch got processed.
     assert_eq!(
+        codes_len,
         listener.message_processed_batch(mids).await?.len(),
-        PATHS.len()
     );
 
     // Checking that blocks still running.
     assert!(listener.blocks_running().await?);
 
     Ok(())
 }
+
+#[tokio::test]
+async fn harmless_upload() -> Result<()> {
+    let mut codes = vec![];
+    for path in &PATHS {
+        codes.push(gclient::code_from_os(path)?);
+    }
+
+    // Creating gear api.
+    //
+    // By default, login as Alice, than re-login as Bob.
+    let api = GearApi::dev().await?.with("//Bob")?;
+
+    upload_programs_and_check(&api, codes).await?;
+
+    Ok(())
+}
+
+#[test]
+fn alloc_zero_pages() {
```

Comment:
I guess it could be run in an async context.

Reply:
either test can be re-written or `block_stopped` checking function

Reply:
with tokio::test, exactly

Reply:
Done